lib/AdobeMarketingCloud/HttpClient.php

```diff
@@ -206,12 +206,16 @@ public function getLastResponse()
      *
      * @return  array   the response
      */
-    protected function decodeResponse($response)
+    protected function decodeResponse($response, $options = array())
     {
-        switch ($this->options['format'])
+        if (count($options) <= 0) {
+            $options = $this->options;
+        }
+
+        switch ($options['format'])
```

Comment:
Do we need to add a default for the `format` switch or add other `format` cases?

Reply:
The options array is initialized to make 'json' the default. If some other format is specified, it throws an exception. If for some reason no decoding format is specified (i.e., the constructor is changes), I think it makes sense to continue to throw an exception here.